pkg/repository/v1/trigger.go

```diff
@@ -116,6 +119,61 @@ type TriggerFromEventsResult struct {
 	EventExternalIdToRuns map[string][]*Run
 }
 
+type TriggerDecision struct {
+	ShouldTrigger bool
+	FilterPayload []byte
+}
+
+func (r *TriggerRepositoryImpl) makeTriggerDecision(ctx context.Context, filters []*sqlcv1.V1Filter, opt EventTriggerOpts) TriggerDecision {
+	if len(filters) == 0 {
+		// If no filters were found, we should trigger the workflow
+		return TriggerDecision{
+			ShouldTrigger: true,
+			FilterPayload: nil,
+		}
+	}
+
+	for _, filterPtr := range filters {
+		if filterPtr == nil {
+			continue
+		}
+
+		filter := *filterPtr
```

Comment:
do I need to do this or can I just refer to the pointer directly?

Reply:
it's fine to make a copy of the pointer `filter := filterPtr`, but in new versions of Go this isn't necessary since each iteration of the loop gets `filterPtr` reassigned to a new value. This didn't used to be the case in < 1.22 or something like that, so if anyone ever decides to compile against an older version (cough cough PR bots) it's still ok to do this. hopefully don't need to worry about this in a few years